docs/junit-extensions.md

```diff
@@ -1,4 +1,36 @@
-#JUnit Extensions
+##JUnit Extensions
+
+JUnit extensions are divided in new method rules, method rule annotations, and test runners.
+
+### JUnit Method Rules
+
+#### MethodRuleAnnotationProcessor
```

Comment:
❓Could we mimic JUnit 5's `@ExtendWith` and the general way JUnit 5 works instead of having a completely different annotation and design?

Reply:
See above answer.